cmd/cluster-display/main.go

```diff
@@ -169,18 +279,44 @@ func main() {
 	if err != nil {
 		logrus.WithError(err).Fatal("could not load kube config")
 	}
-	hiveConfig, ok := kubeConfigs[string(api.HiveCluster)]
-	if len(kubeConfigs) != 1 || !ok {
-		logrus.WithError(err).Fatalf("found %d contexts in Hive kube config and it must be %s", len(kubeConfigs), string(api.HiveCluster))
+
+	inClusterConfig, hasInClusterConfig := kubeConfigs[kube.InClusterContext]
+	delete(kubeConfigs, kube.InClusterContext)
+	delete(kubeConfigs, kube.DefaultClusterAlias)
+
+	if _, hasAppCi := kubeConfigs[appCIContextName]; !hasAppCi {
+		if !hasInClusterConfig {
+			logrus.WithError(err).Fatalf("had no context for '%s' and loading InClusterConfig failed", appCIContextName)
```

Comment:
can we replace all these logging calls to a logger with fields?

Reply:
Updated.